libs/design-system/src/components/ThemeRadio.tsx

```diff
@@ -31,7 +31,22 @@ type Props = {
 }
 
 export function ThemeRadio({ className, tooltipClassName, tabIndex }: Props) {
+  const [shouldMount, setShouldMount] = useState(false)
   const { theme, setTheme } = useTheme()
+
+  // This effect, the shouldMount state above, and the if check below avoids
+  // an unintuitive hydration warning with NextJS by making sure this component
+  // *only* renders on the client side, as we intend with the 'use client'
```

Comment:
Map of rendering behaviour:
- no directive: server-only
- 'use client': runs on both client and server (unintuitive name)
- your change: client-only


The daemon apps all ship as client-side only apps since they have to be exported and served as static files. Their root component has a similar check. We can use this existing component:
https://github.com/SiaFoundation/web/blob/1cd992a28b2128f158bc04ec37045b48d49556d5/libs/design-system/src/site/NextAppCsr.tsx#L9-L10
The benefit beyond reuse being the intent becomes very clear when you see something wrapped with `<ClientSideOnly>`.

We should move it into its own file in src/components, remove the wrapping div, and I think renaming it `ClientSideOnly` would be more clear.

Reply:
I have pushed changes to reflect this. I did have one thought while doing it, though, that is true for the local solution I had and this new implementation you suggested:

We will run into cumulative layout shift issues down the road with this, and the wrapping nature of this component may actually make it tougher to combat. I think this works fine for now, but, in the future, we may want to actually make this a hook that serves up "shouldMount" and then lets the component decide what to do in the false case.  This ThemeRadio component is actually a great example. In the footer, we see the USD dropdown start from right and then get pushed out to left. With a hook like I'm thinking about, the ThemeRadio component could invoke this and then have the responsibility of managing its own placeholder. I can't think of other ways to do it, but there certainly may be.

Reply:
You could add a fallback prop that is rendered instead of null.